pkg/instancegroups/instancegroups.go

```diff
@@ -102,6 +102,64 @@ func promptInteractive(upgradedHostId, upgradedHostName string) (stopPrompting b
 // TODO: Remove from ASG first so status is immediately updated?
 // TODO: Batch termination, like a rolling-update
 
+func (r *RollingUpdateInstanceGroup) CordonUpdate(rollingUpdateData *RollingUpdateCluster, instanceGroupList *api.InstanceGroupList, isCordon bool) (err error) {
+
+	// we should not get here, but hey I am going to check.
+	if rollingUpdateData == nil {
+		return fmt.Errorf("rollingUpdate cannot be nil")
+	}
+
+	// Skip cordoning if cloudonly
+	if rollingUpdateData.CloudOnly {
+		return nil
+	}
+
+	// Do not need a k8s client if you are doing cloudonly.
+	if rollingUpdateData.K8sClient == nil && !rollingUpdateData.CloudOnly {
+		return fmt.Errorf("cordonUpdater is missing a k8s client")
+	}
+
+	if instanceGroupList == nil {
+		return fmt.Errorf("cordonUpdater is missing the InstanceGroupList")
+	}
+
+	// add the instances that require updates to this list
+	update := r.CloudGroup.NeedUpdate
+	// if --force is used, add nodes to update list anyway
+	if rollingUpdateData.Force {
+		update = append(update, r.CloudGroup.Ready...)
+	}
+
+	if len(update) == 0 {
+		return nil
+	}
+
+	if featureflag.CordonAllWhenRollingUpdate.Enabled() {
+		if rollingUpdateData.Cordon {
+			for _, u := range update {
+				nodeName := ""
+				if u.Node != nil {
+					nodeName = u.Node.Name
+				}
+				klog.Infof("Cordoning the node: %q.", nodeName)
+
+				err = r.CordonNode(u, rollingUpdateData, isCordon)
+				if err != nil {
+					if rollingUpdateData.FailOnDrainError {
+						// r.uncordonOnExit(rollingUpdateData, isBastion)
+						return fmt.Errorf("failed to cordon node %q: %v", nodeName, err)
```

Comment:
It could have failed uncordoning the node, could it not?
If it failed uncordoning a node, perhaps because something else deleted it, returning an error here will mean all the rest of the nodes in `range update` will be incorrectly left cordoned.